### PR TITLE
Scoping `icu_locid` macros nicely

### PR DESCRIPTION
**Pull request by robertbastian**
Figured out how to put them into the submodules.

### CODE REVIEW

File: components/locid/src/helpers.rs [hunk 1]
```diff
@@ -458,37 +459,40 @@ macro_rules! impl_tinystr_subtag {
...
                         Ok(r) => r,
                         #[allow(clippy::panic)] // const context
-                        _ => panic!(concat!("Invalid ", stringify!($name), ": ", $string)),
+                        _ => panic!(concat!("Invalid ", $(stringify!($path), "::",)+ stringify!($name), ": ", $string)),
                     };
                 R
             }};
         }
+        #[doc(inline)]
+        pub use $legacy_macro_name as $macro_name;
```
Comment (sffc) [inline components/locid/src/helpers.rs:488]: Praise: Seems so obvious in hindsight; and I guess it works in Rust 1.61 or else CI would fail

**Review by sffc — APPROVED** [review]
Would prefer a review from @zbraniecki as well but feel free to merge if this is blocking anything

File: components/locid/src/helpers.rs (hunk 1 above)
Comment (robertbastian) [reply components/locid/src/helpers.rs:488]: This has probably been possible since 1.0.

Comment (Manishearth) [reply components/locid/src/helpers.rs:488]: haven't we been using this trick already? I definitely recall mentioning it and I think baked_exporter uses it a bunch

**Review by Manishearth — APPROVED** [review]

File: components/locid/src/helpers.rs (hunk 1 above)
Comment (robertbastian) [reply components/locid/src/helpers.rs:488]: We thought this wasn't possible, see https://github.com/unicode-org/icu4x/issues/253#issuecomment-1133207454. I figure out how to do this for baked exporter, and now am applying it here.

**Comment by sffc** [comment]
I wonder if we should keep the fully qualified names as non-hidden API. We can't avoid them, so we may as well embrace them as an alternative import.

**Comment by robertbastian** [comment]
We can avoid them being documented, like many other place that use `#[doc(hidden)]`. I'd rather have a canonical doc location for each item, and I don't want to mark the module locations as reexports.

**Comment by sffc** [comment]
Does doc(no_inline) or similar work?

**Review by sffc — APPROVED** [review]

**Pull request merged**